packages/babel-plugin-transform-es2015-template-literals/src/index.js

```diff
@@ -45,23 +51,27 @@ export default function ({ types: t }) {
           if (index < expressions.length) {
             const expr = expressions[index++];
             const node = expr.node;
-            if (state.opts.spec && !expr.isBaseType("string") && !expr.isBaseType("number")) {
-              nodes.push(t.callExpression(t.identifier("String"), [node]));
-            } else if (!t.isStringLiteral(node, { value: "" })) {
+            if (!t.isStringLiteral(node, { value: "" })) {
               nodes.push(node);
             }
           }
         }
 
         // since `+` is left-to-right associative
         // ensure the first node is a string if first/second isn't
-        if (!t.isStringLiteral(nodes[0]) && !t.isStringLiteral(nodes[1])) {
+        if (!t.isStringLiteral(nodes[0]) && (state.opts.spec || !t.isStringLiteral(nodes[1]))) {
           nodes.unshift(t.stringLiteral(""));
         }
+        let root = nodes.shift();
```

Comment:
I just got rid of this `#shift` 😛. Let's do:

```js
let root = nodes[0];
if (spec && nodes.length > 1) {
  root = t.callExpression(t.memberExpression(target, t.identifier("concat")), nodes.slice(1));
} else {
  for (;;) {
    // ...
  }
```

Whether you use the helper function is up to you.